xrtpy/response/xrt_teem.py

```diff
@@ -0,0 +1,428 @@
+import numpy as np
+import sys
+
+from astropy import units as u
+from astropy.constants import c, h
+from sunpy.coordinates.sun import angular_radius
+
+from xrtpy.response.temperature_response import TemperatureResponseFundamental
+
+
+def xrt_teem(
+    map1,
+    map2,
+    binfac=1,
+    Trange=None,
+    no_threshold=False,
+    Te_err_threshold=0.5,
+    photon_noise_threshold=0.2,
+    mask=None,
+    verbose=False,
+):
+    r"""
+    Get coronal temperatures and emission measures from a pair of images using
+    the filter ratio method.
+
+    Currently this program uses the solar spectrum calculated with CHIANTI
+    database ver. 6.0.1 (density: :math:`10^9` cm\ :sup:`-3`\ , ionization
+    equilibrium: chianti.ioneq, abundance: sun_coronal_ext), because this
+    is the only spectrum available in xrtpy. We expect this to change.
+
+    Parameters:
+    -----------
+    map1 : XRTMap object (i.e. sunpy.map.sources.hinode.XRTMap)
+        map for the first XRT level 1 data image.  If the image is
+        normalized, then it is assumed that the un-normalized image can be
+        recovered by multiplying by the exposure time (exposure time must be
+        available in the metadata). It is also assumed that the metadata
+        history will contain the string XRT_RENORMALIZE if the image has been
+        normalized.
+
+    map2 : XRTMap object
+        map for the second image (must use different filters from the first
+        image). The image shape should match that in map1. The same
+        considerations apply as for map1.
+
+    binfac : integer, optional (default = 1)
+        spatial binning factor
+
+    Trange : 2 element sequence containing floats [Optional]
+        Range of log10(temperature) values to examine. Must be in order from
+        lower to higher.
+
+    Te_err_threshold : float [Optional]
+        Threshold value for the temperature error. (default = 0.5 [50%])
+        Values of temperature with error exceeding the threshold are set to 0.
+
+    photon_noise_threshold : float [Optional]
+        Threshold value for ratio of photon noise to signal. (default = 0.2
+        [20%]). If photon noise/signal exceeds this value, the corresponding
+        temperatures are set to 0.
+
+    mask : Boolean array of the same shape as the images [Optional]
+        If provided, masks out parts of the images from the analysis. Note:
+        pixels to be masked out should be True, unmasked should be False
```

Comment:
Each sunpy `Map` can have an associated mask [(see this example)](https://docs.sunpy.org/en/stable/generated/gallery/map/image_bright_regions_gallery_example.html) so it may not be necessary to pass this in separately. For example, the mask the resulting temperature and emission measure maps could just be the union of the two input masks.